modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java

```diff
@@ -137,4 +139,55 @@
      * @return The current value
      */
     T getValue();
+
+    /**
+     * Returns an {@link ObservableValue} which provides a mapping of the value
+     * held by this {@code ObservableValue}, and provides {@code null} when this
+     * {@code ObservableValue} holds {@code null}.
+     *
+     * @param <U> the type of values held by the resulting {@code ObservableValue}
+     * @param mapper a {@link Function} which converts a given value to a new value, cannot be null
+     * @return an {@link ObservableValue} which provides a mapping of the value
+     *     held by this {@code ObservableValue}, and provides {@code null} when
+     *     this {@code ObservableValue} holds {@code null}, never null
+     */
+    default <U> ObservableValue<U> map(Function<? super T, ? extends U> mapper) {
+        return new MappedBinding<>(this, mapper);
+    }
+
+    /**
+     * Returns an {@link ObservableValue} which provides a mapping of the value
+     * held by this {@code ObservableValue}, or {@code constant} when this
+     * {@code ObservableValue} holds {@code null}.
+     *
+     * @param constant an alternative value to use when this {@code ObservableValue}
+     *     holds {@code null}, can be null
+     * @return an {@link ObservableValue} which provides a mapping of the value
+     *     held by this {@code ObservableValue}, or {@code constant} when this
+     *     {@code ObservableValue} holds {@code null}, never null
+     */
+    default ObservableValue<T> orElse(T constant) {
+        return new OrElseBinding<>(this, constant);
+    }
+
+    /**
+     * Returns an {@link ObservableValue} which provides the value in the {@code
+     * ObservableValue} given by applying {@code mapper} on the value held by this
+     * {@code ObservableValue}, and is {@code null} when this
+     * {@code ObservableValue} holds {@code null}.<p>
+     *
+     * Returning {@code null} from {@code mapper} will result in an
+     * {@code ObservableValue} which holds {@code null}.
```

Comment:
This one is hard for me to explain. It comes out convoluted regardless of how I phrase it.

```
Creates an {@code ObservableValue} that holds the value of the {@code ObservableValue} resulting
from applying a mapping on the value held by this {@code ObservableValue}. The result is updated (lazily)
when either the values held by this {@code ObservableValue} or the value held by the {@code ObservableValue}
in the mapping changes. If this value or the value returned by the mapping are {@code null}, the held value
is also {@code null}.
<p>
For example, a property which is only true when the UI is visible:
<pre>
    ObservableValue<Boolean> allShowing = listView.sceneProperty()
        .flatMap(Scene::windowProperty)
        .flatMap(Window::showingProperty)
        .orElse(false);
</pre>
Changes in the values of any of: the scene of {@code listView}, the window of that scene, or the showing of that
window, will updated the result boolean value {@code allShowing}.
<p>
This method is preferred over {@link javafx.beans.binding.Bindings Bindings#select} methods since it is type
safe and handles {@code null}s in the call chain more easily.
```